params/include/alps/params/params_impl.hpp

```diff
@@ -219,6 +219,12 @@ namespace alps {
                 throw option_type::visitor_none_used("Attempt to use uninitialized option value");
             }
 
+            /// Even though this overload must be defined, it should never be called
+            void operator()(const boost::optional<detail::trigger_tag>& val) const
+            {
+              assert(false);
```

Comment:
propose to throw rather than assert so it doesn't survive the debug mode, either.

Reply:
You mean the release mode, right?

Reply:
yes, sorry.